modules/environment/login/login.nix

```diff
@@ -0,0 +1,37 @@
+# Licensed under GNU Lesser General Public License v3 or later, see COPYING.
+# Copyright (c) 2019 Alexander Sosedkin and other contributors, see AUTHORS.
+
+{ config, writeScript }:
+
+let
+  inherit (config.build) installationDir;
+in
+
+writeScript "login" ''
+  #!/system/bin/sh
+  set -e
+
+  export USER="${config.user.userName}"
+  export PROOT_TMP_DIR=${installationDir}/tmp
+  export PROOT_L2S_DIR=${installationDir}/.l2s
+
+  if test -e ${installationDir}/bin/.proot-static.new; then
```

Comment:
In the activation script, I am installing `proot-static`, `login` and `login-inner`. My first attempt was to replace these files directly on activation. When trying to replace `proot-static`, I got a "file is busy" error message. When trying to replace `login-inner`, the execution of `login-inner` stopped immediately (after finishing the started child processes like in this case `nix-on-droid switch`). Replacing `login` works without problems.

To clarify: These problems occur everytime I install a new version of any of these files, not just on the initial boot.

The installation of the new files need to be at login time, because there I can safely replace the old files.

As I write this, I think I forgot the case when there are multiple running sessions. I need to somehow include a check, if there is no other session running and only then replace these files..

Reply:
If we do not need to support the old setup the broken symlinks logic can be removed from the app.

BTW, the problem with multiple sessions has been fixed